legal-api/src/legal_api/reports/report.py

```diff
@@ -586,8 +587,36 @@ def _format_consent_continuation_out_data(self, filing):
             self._format_address(filing['offices']['registeredOffice']['mailingAddress'])
 
     def _format_agm_extension_data(self, filing):
-        # FUTURE: format logic for letter of agm extension
-        return
+        meta_data = self._filing.meta_data or {}
+        is_first_agm = meta_data.get('agmExtension', {}).get('isFirstAgm', '')
+        filing['is_first_agm'] = is_first_agm
+        filing['agm_year'] = meta_data.get('agmExtension', {}).get('year', '')
+        filing['is_final_agm'] = meta_data.get('agmExtension', {}).get('isFinalExtension', '')
+
+        number_words = ['one', 'two', 'three', 'four', 'five', 'six']
+        duration_numeric = meta_data.get('agmExtension', {}).get('extensionDuration', '')
+        filing['duration_numeric'] = duration_numeric
+        filing['duration_spelling'] = number_words[int(duration_numeric)-1]
+
+        if is_first_agm:
+            founding_date_json = self._filing.filing_json['filing'].get('business', {}).get('foundingDate', '')
+            founding_date = founding_date_json[0:10]
+            original_date_time = LegislationDatetime.\
+                as_legislation_timezone_from_date_str(founding_date) + relativedelta(months=18)
+            filing['original_agm_date'] = original_date_time.strftime(OUTPUT_DATE_FORMAT)
+        else:
+            expire_date_current_string = meta_data.get('agmExtension', {}).get('expireDateCurrExt', '')
+            date_current_obj = LegislationDatetime.as_legislation_timezone_from_date_str(expire_date_current_string)
+            filing['original_agm_date'] = date_current_obj.strftime(OUTPUT_DATE_FORMAT)
+
+        expire_date_approved_string = meta_data.get('agmExtension', {}).get('expireDateApprovedExt', '')
+        date_approved_obj = datetime.strptime(expire_date_approved_string, '%Y-%m-%d')
```

Comment:
Although it should be impossible to get an empty string in each approved scenario, I think it's better to check it before doing datetime conversion. Can do something like this https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/reports/report.py#L514-L515

Reply:
There're some optional fields in `meta_data` ^^